src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/SQLiteItemWriter.java

```diff
@@ -0,0 +1,216 @@
+package org.openstreetmap.atlas.geography.atlas.sqlite;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Rectangle;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasItem;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.ItemType;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedTagStore;
+import org.openstreetmap.atlas.geography.converters.WkbLocationConverter;
+import org.openstreetmap.atlas.utilities.maps.LongToLongMap;
+
+/**
+ * To save all of the data within an atlas, we need to get each {@link AtlasItem} type and iterate
+ * though them with inserts into the database. This class declares that a given writer needs to get
+ * the atlas to do its job.
+ *
+ * @author hallahan
+ * @author lcram
+ */
+public class SQLiteItemWriter
+{
+    private static final String INSERT_SQL = "INSERT INTO %s VALUES (?, ?, ?, ?)";
+    private static final String INSERT_TAGS_SQL = "INSERT INTO %s_tags VALUES (?, ?, ?)";
+    private static final String INSERT_RTREE_SQL = "INSERT INTO %s_rtree VALUES (?, ?, ?, ?, ?)";
+
+    // database column constants
+    private static final int FIRST_COLUMN = 1;
+    private static final int SECOND_COLUMN = 2;
+    private static final int THIRD_COLUMN = 3;
+    private static final int FOURTH_COLUMN = 4;
+    private static final int FIFTH_COLUMN = 5;
```

Comment:
I used to do that, but it get's a bit hard to maintain eventually. ++column is easier. But, I also am fine with whatever.

Reply:
Would it be preferable to just add a checkstyle exception for this class, specifically for the magic constants rule? That way we could just use integer literals. Or do you still think `++column` is easier to maintain? I did find that you have to be careful about resetting it to `0` all over the place.

Reply:
I have no preference.